src/Digdir.Domain.Dialogporten.Application/Common/IUserNameRegistry.cs

```diff
@@ -18,24 +19,49 @@ public class UserNameRegistry : IUserNameRegistry
 {
     private readonly IUser _user;
     private readonly INameRegistry _nameRegistry;
+    private readonly IOrganizationRegistry _organizationRegistry;
 
-    public UserNameRegistry(IUser user, INameRegistry nameRegistry)
+    public UserNameRegistry(IUser user, INameRegistry nameRegistry, IOrganizationRegistry organizationRegistry)
     {
         _user = user ?? throw new ArgumentNullException(nameof(user));
         _nameRegistry = nameRegistry ?? throw new ArgumentNullException(nameof(nameRegistry));
+        _organizationRegistry = organizationRegistry ?? throw new ArgumentNullException(nameof(organizationRegistry));
     }
 
-    public bool TryGetCurrentUserPid([NotNullWhen(true)] out string? userPid) => _user.TryGetPid(out userPid);
+    public bool TryGetCurrentUserExternalId([NotNullWhen(true)] out string? userExternalId)
+    {
+        if (_user.TryGetPid(out userExternalId)) return true;
+        if (_user.TryGetLegacySystemUserId(out userExternalId)) return true;
+        if (_user.TryGetOrgNumber(out userExternalId)) return true;
+        return false;
+    }
 
     public async Task<UserInformation?> GetUserInformation(CancellationToken cancellationToken)
     {
-        if (!TryGetCurrentUserPid(out var userPid))
+        if (!TryGetCurrentUserExternalId(out var userExernalId))
         {
             return null;
         }
 
-        var userName = await _nameRegistry.GetName(userPid, cancellationToken);
-        return new(userPid, userName);
+        string? userName;
+        switch (_user.GetPrincipal().GetUserType())
+        {
+            case UserType.Person:
+                userName = await _nameRegistry.GetName(userExernalId, cancellationToken);
+                break;
+            case UserType.LegacySystemUser:
+                _user.TryGetLegacySystemUserName(out userName);
+                break;
+            case UserType.Enterprise:
+                userName = await _organizationRegistry.GetOrgShortName(userExernalId, cancellationToken);
+                break;
+            case UserType.Unknown:
+            case UserType.SystemUser: // Implement when we know how this will be handled
+            default:
+                throw new UnreachableException("Unknown user type");
```

Comment:
Potensiell endring i behaviour her men, det er vel noe vi har på planene å ha felles sjekker lenger opp i chainen for validering av tokens?

Reply:
Mhm å finne usertype er nok noe som kunne vært gjort i en authN-mellomvare, og så bare satt som en claim (evt kasta 401). Så hadde vi sluppet alle handler-sjekkene. Skal se på det.